src/routes/profile.js

```diff
@@ -53,132 +55,113 @@ router.get('/mod/customavatar', isModMiddleware, (req, res) => {
+      await userRequestingAvatar.save();
 
-      const modUser = req.user;
-      // Create mod log - Doesn't need to be async
-      ModLog.create({
-        type: 'avatar',
-        modWhoMade: {
-          id: modUser._id,
-          username: modUser.username,
-          usernameLower: modUser.usernameLower,
-        },
-        data: {
-          modComment: req.body.modcomment,
-          approved: req.body.decision,
-          username: foundReq.forUsername,
-          msgToMod: foundReq.msgToMod,
-          resLink: foundReq.resLink,
-          spyLink: foundReq.spyLink,
-        },
-        dateCreated: new Date(),
-      });
+      // TODO-kev: Fundamental flaw in createNotification. Passing non unique link #
+      let str = `Your avatar request was approved by ${modWhoProcessed.username}! Their comment was: ${avatarReq.modComment}`;
+      createNotification(
+        userRequestingAvatar._id,
+        str,
+        '#',
+        modWhoProcessed.username,
+      );
+    } else if (req.body.decision === false || req.body.decision === 'false') {
+      await s3.rejectAvatarRequest(avatarReq.resLink);
+      await s3.rejectAvatarRequest(avatarReq.spyLink);
+
+      avatarReq.resLink = null;
+      avatarReq.spyLink = null;
+      avatarReq.markModified('resLink');
+      avatarReq.markModified('spyLink');
+
+      let str = `Your avatar request was rejected by ${modWhoProcessed.username}. Their comment was: ${avatarReq.modComment}`;
+      createNotification(
+        userRequestingAvatar._id,
+        str,
+        '#',
+        modWhoProcessed.username,
+      );
+    } else {
+      // TODO-kev: Change the error?
+      console.log(
+        `error, decision isnt anything recognisable...: ${req.body.decision}`,
+      );
+      return;
+    }
 
-      await foundReq.save();
+    // Create mod log - Doesn't need to be async
+    ModLog.create({
+      type: 'avatar',
+      modWhoMade: {
+        id: modWhoProcessed._id,
+        username: modWhoProcessed.username,
+        usernameLower: modWhoProcessed.usernameLower,
+      },
+      data: {
+        modComment: req.body.modcomment,
+        approved: req.body.decision,
+        username: avatarReq.forUsername,
+        msgToMod: avatarReq.msgToMod,
+        resLink: avatarReq.resLink,
+        spyLink: avatarReq.spyLink,
+      },
+      dateCreated: new Date(),
+    });
 
-      if (req.body.decision === false || req.body.decision === 'false') {
-        await foundReq.remove();
-      }
+    if (req.body.decision === false || req.body.decision === 'false') {
+      await avatarReq.remove();
+    } else {
+      avatarReq.processed = true;
+      avatarReq.modComment = req.body.modcomment;
+      avatarReq.approved = req.body.decision;
+      avatarReq.modWhoProcessed = modWhoProcessed.username;
+      avatarReq.markModified('processed');
+      avatarReq.markModified('modComment');
+      avatarReq.markModified('approved');
+      avatarReq.markModified('modWhoProcessed');
```

Comment:
Let's also try removing these mark modified calls as well as they shouldn't be needed.